tests/test_mc_comm.py

```diff
@@ -0,0 +1,229 @@
+
+
+def test_McCommunicationProcess_register_magic_word__registers_with_magic_word_in_serial_comm_from_board__when_first_packet_is_truncated_to_less_than_8_bytes__and_calls_read_with_correct_size__and_calls_sleep_correctly(
+    four_board_mc_comm_process, mantarray_mc_simulator_no_beacon, mocker
+):
+    # mock sleep to speed up the test
+    mocked_sleep = mocker.patch.object(mc_comm, "sleep", autospec=True)
+
+    mc_process = four_board_mc_comm_process[0]
+    simulator = mantarray_mc_simulator_no_beacon[4]
+
+    # need to mock read here to have better control over the reads going into McComm
+    test_read_values = [
+        SERIAL_COMM_MAGIC_WORD_BYTES[:4],
+        bytes(0),
+        bytes(0),
+        bytes(0),
+        bytes(0),
+        SERIAL_COMM_MAGIC_WORD_BYTES[4:],
+    ]
+    mocked_read = mocker.patch.object(
+        simulator, "read", autospec=True, side_effect=test_read_values
+    )
+
+    board_idx = 0
+    invoke_process_run_and_check_errors(mc_process)
+    assert mc_process.is_registered_with_serial_comm(board_idx) is False
+
+    mc_process.set_board_connection(board_idx, simulator)
+    invoke_process_run_and_check_errors(mc_process)
+    assert mc_process.is_registered_with_serial_comm(board_idx) is True
+
+    # Assert it reads once initially then once per second until status beacon period is reached (a new packet should be available by then)
+    assert (
+        len(mocked_read.call_args_list) == SERIAL_COMM_STATUS_BEACON_PERIOD_SECONDS + 1
+    )
+    assert mocked_read.call_args_list[0] == mocker.call(size=8)
+    assert mocked_read.call_args_list[1] == mocker.call(size=4)
+    assert mocked_read.call_args_list[2] == mocker.call(size=4)
+    assert mocked_read.call_args_list[3] == mocker.call(size=4)
+    assert mocked_read.call_args_list[4] == mocker.call(size=4)
+
+    # Assert sleep is called with correct value and correct number of times
+    expected_sleep_secs = 1
+    for sleep_call_num in range(SERIAL_COMM_STATUS_BEACON_PERIOD_SECONDS - 1):
+        sleep_iter_call = mocked_sleep.call_args_list[sleep_call_num][0][0]
+        assert (sleep_call_num, sleep_iter_call) == (
+            sleep_call_num,
+            expected_sleep_secs,
+        )
+
+
+def test_McCommunicationProcess_register_magic_word__raises_error_if_less_than_8_bytes_available_after_status_beacon_period_has_elapsed(
+    four_board_mc_comm_process, mantarray_mc_simulator_no_beacon, mocker
+):
+    mocker.patch(
+        "builtins.print", autospec=True
+    )  # don't print all the error messages to console
+
+    # mock sleep to speed up the test
+    mocker.patch.object(mc_comm, "sleep", autospec=True)
+
+    mc_process = four_board_mc_comm_process[0]
+    simulator = mantarray_mc_simulator_no_beacon[4]
+
+    # need to mock read here to have better control over the reads going into McComm
+    test_read_values = [
+        SERIAL_COMM_MAGIC_WORD_BYTES[:-1],
+        bytes(0),
+        bytes(0),
+        bytes(0),
```

Comment:
why are there 5x0bytes here? is it related to some constant? (sleeping for 5 seconds?)